compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -640,6 +640,34 @@ impl<'a> Parser<'a> {
             }
         }
 
+        // Extra info for `c"str"` before 2021 edition or `c "str"` in all editions. The heuristic
```

Comment:
```suggestion
        // Try to detect an attempt by the user to write a c-string literal before the 2021 edition. The heuristic
```

or something like that (it gives a better summary imo). Note that your comment still mentions `c "str"` (with a space) which is no longer accurate.